org.eclipse.scout.rt.dataobject/src/main/java/org/eclipse/scout/rt/dataobject/id/IdInventory.java

```diff
@@ -96,4 +128,31 @@ public String getTypeName(IId id) {
   public Class<? extends IId> getIdClass(String typeName) {
     return m_nameToClassMap.get(typeName);
   }
+
+  /**
+   * @return Whether the {@link IId} needs to be serialized and deserialized using encryption or not. For more details
+   *         see {@link IdEncryption}.
+   */
+  public boolean isIdEncryption(Class<? extends IId> idClass) {
```

Comment:
Two suggestions:

- change to concurrent hash map and remove synchronized block
- for me recursive stream API code is pretty complex to read. suggenstion: use computeIfAbsent value resolve method which is implemented using classic style code (for me I see no advantages in using stream-based coding here as the code requires a lot of comments to be understanable when reading)

Suggestion

```
  public boolean isIdEncryption(Class<? extends IId> idClass) {
    return m_classToIdEncryptionMap.computeIfAbsent(idClass, this::computeIdSignature);
  }
  
  protected boolean computeIdSignature(Class<? extends IId> idClazz) {
    Class<?> clazz = idClazz;
    while (true) {
      IdEncryption idEncryption = clazz.getAnnotation(IdEncryption.class);
      if (idEncryption != null) {
        return idEncryption.value();
      }
      if (clazz == null || clazz == Object.class) {
        return false;
      }
      clazz = clazz.getSuperclass();
    }
  }
}
````

Reply:
This implementation will only compute the value for the requested `Class<? extends IId>` even if there is a large super class hierarchy. My goal was to collect all results of the super class hierarchy. Doing this with `computeIfAbsent` will result in `ConcurrentModificationException`s. Therefore, all results need to be collected or added in a recursion, in both options we need to synchronize explicitly.